crates/host/src/wasmbus/mod.rs

```diff
@@ -2355,6 +2403,42 @@ impl Host {
         }
     }
 
+    async fn store_actor_claims(&self, claims: jwt::Claims<jwt::Actor>) -> anyhow::Result<()> {
+        self.store_claims(
+            claims
+                .try_into()
+                .context("failed to convert actor claims")?,
+        )
+        .await
+    }
+
+    async fn store_provider_claims(
+        &self,
+        claims: jwt::Claims<jwt::CapabilityProvider>,
+    ) -> anyhow::Result<()> {
+        self.store_claims(
+            claims
+                .try_into()
+                .context("failed to convert provider claims")?,
+        )
+        .await
+    }
+
+    async fn store_claims(&self, stored_claims: StoredClaims) -> anyhow::Result<()> {
```

Comment:
Why not take `impl TryInto<StoredClaims>` as argument instead and remove both `store_actor_claims` and `store_provider_claims` ?

Reply:
I couldn't get this to work earlier, but you got me to go back to it 👍 `impl TryInto<StoredClaims>` isn't enough, but this is:
```T: TryInto<StoredClaims, Error = anyhow::Error>```